packages/amplify-graphql-model-transformer/src/resources/amplify-dynamodb-table/amplify-table-manager-lambda/amplify-table-manager-handler.ts

```diff
@@ -25,24 +30,100 @@ const notFinished: AWSCDKAsyncCustomResource.IsCompleteResponse = {
   IsComplete: false,
 };
 
+// #region Entry points
+export const onEvent = cfnResponse.safeHandler(onEventHandler);
+export const isComplete = cfnResponse.safeHandler(isCompleteHandler);
 /**
- * Util function to log especially for nested objects and arrays
- * @param msg
- * @param other arrays of arguments to be logged
+ * Handler for requests sent from CloudFormation for custom resource.
+ * This is the entry point of the custom resource.
+ *
+ * Note: This is adapted from the AWS CDK's provider framework.
+ * https://github.com/aws/aws-cdk/blob/11621e7/packages/aws-cdk-lib/custom-resources/lib/provider-framework/runtime/framework.ts
+ * @param cfnRequest Request received from CloudFormation
  */
-const log = (msg: string, ...other: any[]) => {
-  console.log(
-    msg,
-    other.map((o) => (typeof o === 'object' ? JSON.stringify(o, undefined, 2) : o)),
-  );
-};
+// eslint-disable-next-line func-style
+async function onEventHandler(cfnRequest: AWSLambda.CloudFormationCustomResourceEvent): Promise<void> {
+  const sanitizedRequest = { ...cfnRequest, ResponseURL: '...' } as const;
+  log('onEventHandler', sanitizedRequest);
+
+  cfnRequest.ResourceProperties = cfnRequest.ResourceProperties || {};
+
+  const onEventResult = await processOnEvent(sanitizedRequest);
+  log('onEvent returned:', onEventResult);
+
+  // merge the request and the result from onEvent to form the complete resource event
+  // this also performs validation.
+  const resourceEvent = createResponseEvent(cfnRequest, onEventResult);
+  log('event:', onEventResult);
```

Comment:
1. Did you intend to log `resourceEvent` rather than `onEventResult`?
2. Do we know `resourceEvent` is safe to log?

Reply:
1. Good catch. The [same issue](https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/custom-resources/lib/provider-framework/runtime/framework.ts#L38) exists in the CDK's custom resource provider today.
2. As-is, no it's not something we want to log because it contains the unredacted `ResponseURL` for the custom resource.

I'm going to remove this log statement, commit incoming.

Reply:
Addressed in 55f150ffdb44fbcaad8f2e48a1cbb879afa5b459